pkg/network/server.go

```diff
@@ -793,7 +802,7 @@ func (s *Server) requestBlocksOrHeaders(p Peer) error {
 	}
 	err := s.requestBlocks(bq, p)
 	if err != nil {
-		return err
+		return fmt.Errorf("%w: failed to request blocks: %w", errInvalidBlock, err)
```

Comment:
It's not "invalid block", it's literally "blocks request failed", so create a separate error with "block request failed" text and use `return fmt.Errorf("%w: %w", errBlocksRequestFailed, err)` here.